middleware.go

```diff
@@ -211,6 +212,60 @@ func (r *oauthProxy) authenticationMiddleware(resource *Resource) func(http.Hand
 	}
 }
 
+// checkClaim checks whether claim in userContext matches claimName, match. It can be String or Strings claim.
+func (r *oauthProxy) checkClaim(user *userContext, claimName string, match *regexp.Regexp, resourceURL string) bool {
+	errFields := []zapcore.Field{
+		zap.String("claim", claimName),
+		zap.String("access", "denied"),
+		zap.String("email", user.email),
+		zap.String("resource", resourceURL),
+	}
+
```

Comment:
given [Claims](https://github.com/gambol99/go-oidc/blob/master/jose/claims.go#L10) is `type Claims map[string]interface{}` .. we could just and fail quick .. What do you think?

```
if _, found := user.claims[claimName]; !found {
    r.log.Warn("the token does not have the claim", errFields...)
    return false
}

Reply:
Yep, I've added the statement here.